From: ryanking13 <def6488@gmail.com>
Date: Thu, 13 Aug 2026 19:30:00 +0900
Subject: [PATCH] Disable SO_PEERCRED for Emscripten

Emscripten exposes the SO_PEERCRED constant but does not provide the struct
ucred definition required by PostgreSQL's getpeereid implementation. Skip
this Unix-domain socket credential path when building for Emscripten.

TODO: either remove SO_PEERCRED from emscripten or implement getsockopt(SO_PEERCRED) in Emscripten

---
 src/port/getpeereid.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/port/getpeereid.c b/src/port/getpeereid.c
--- a/src/port/getpeereid.c
+++ b/src/port/getpeereid.c
@@ -32,7 +32,7 @@ int
 getpeereid(int sock, uid_t *uid, gid_t *gid)
 {
-#if defined(SO_PEERCRED)
+#if defined(SO_PEERCRED) && !defined(__EMSCRIPTEN__)
 	/* Linux: use getsockopt(SO_PEERCRED) */
 	struct ucred peercred;
 	socklen_t	so_len = sizeof(peercred);
